sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -1127,6 +1127,57 @@ build_options{{
 Relax the requirement that parameter types for free-function kernels must be
 forward-declarable.
 
+===== `--auto-pch`
+
+Enable auto-detection of the preamble and use it as a pre-compiled header to
+speed up subsequent compilations of TUs matching the preamble/compilation
+options. Example of the code that can benefit from this:
+
+[source,c++]
+----
+#include <sycl/sycl.hpp>
+
+// Auto-detected preamble ends before next line:
+namespace syclext = sycl::ext::oneapi;
+namespace syclexp = sycl::ext::oneapi::experimental;
+
+extern "C"
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void iota(float start, float *ptr) {
+    size_t id = syclext::this_work_item::get_nd_item<1>().get_global_linear_id();
+    ptr[id] = start + static_cast<float>(id);
+}
+----
+
+Limitations:
+
+* Preamble detection is done at the Lexer level and can't handle code like
+
+[source,c++]
+----
+#if 1
+#include <sycl/sycl.hpp>
+#else
+// Auto-detected preamble ends in the middle of `#else` and would fail to compile.
+void foo() {}
+#endif
+----
```

Comment:
This limitation isn't clear to me.  Does the limitation exist whenever there is a `#if / #endif`?  Or, is it a limitation because the preamble ends inside the body of an `#if / #endif`?

How is this handled?  Does the user get an obvious error, so they know it's related to `--auto-pch`?  Can we instead make this work by ending the preamble at the `#if` that encloses the non-directive statement?